### PR TITLE
set https_proxy for vagrant VM's command wget

### PR DESCRIPTION
**Pull request by jomenxiao**
set  https_proxy for wget command

like:
`VM_SET_PROXY=http://127.0.0.1:80/ IPV4=1 K8S=1 NWORKERS=1 contrib/vagrant/start.sh`

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7289)

### CODE REVIEW

File: contrib/vagrant/start.sh [hunk 1]
```diff
@@ -260,6 +269,12 @@ export NWORKERS="${NWORKERS}"
 if [ -z "${RELOAD}" ]; then
     export INSTALL="1"
 fi
+
+if [ -n "${VM_PROXY}" ]; then
+    export WGET="HTTPS_RPXOY=${VM_PROXY} wget"
```
Comment (nebril) [inline contrib/vagrant/start.sh]: `s/HTTPS_RPXOY/HTTPS_PROXY`

File: contrib/vagrant/start.sh [hunk 2]
```diff
@@ -223,6 +226,12 @@ export RUNTIME="${RUNTIME}"
...
```
Comment (nebril) [inline contrib/vagrant/start.sh]: s/HTTPS_RPXOY/HTTPS_PROXY

**Review by nebril — CHANGES_REQUESTED** [review]
Looks great! Please fix the typo, we will run CI after that.

**Review by aanm — COMMENTED** [review]
Thanks for the PR @jomenxiao can you provide a reason for this change?

**Comment by ianvernon** [comment]
test-docs-please

**Comment by jomenxiao** [comment]
@aanm 

- in internal network, access outside by proxy
- in China, we can not reach domain `dl.k8s.io`

**Review by aanm — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Review by nebril — DISMISSED** [review]
Code looks good to me! Please squash the commits into one.

**Comment by nebril** [comment]
test-me-please

File: contrib/vagrant/start.sh [hunk 3]
```diff
@@ -51,6 +51,9 @@ export 'CILIUM_SCRIPT'=true
 # Sets the directory where the temporary setup scripts are created
 export 'CILIUM_TEMP'="${dir}"
 
+# Sets VM's Command wget with HTTPS_PROXY
+export 'VM_PROXY'="${VM_SET_PROXY}"
```
Comment (joestringer) [inline contrib/vagrant/start.sh:55]: This uses `VM_SET_PROXY`, but I don't see that anywhere else in this PR, including in the documentation. Where does it come from?

File: Documentation/contributing.rst [hunk 4]
```diff
@@ -118,6 +118,7 @@ brought up by vagrant:
   cluster. Valid options are: ``docker``, ``containerd`` and ``crio``. If not
   set, it defaults to ``docker``.
 * ``VAGRANT_DEFAULT_PROVIDER={virtualbox \| libvirt \| ...}``
+* ``VM_PROXY=https://127.0.0.1:80/`` Sets up VM's ``https_proxy``.
```
Comment (joestringer) [inline Documentation/contributing.rst]: Is this supposed to be like the following?

```suggestion
* ``VM_SET_PROXY=https://127.0.0.1:80/`` Sets up VM's ``https_proxy``.
```

Comment (nebril) [reply Documentation/contributing.rst]: Good catch, seems like it may be the case

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**